test/platform-routes.test.js

```diff
@@ -0,0 +1,79 @@
+import http from 'node:http'
+import { once } from 'node:events'
+import assert from 'node:assert'
+import pg from 'pg'
+import createDebug from 'debug'
+
+import { assertResponseStatus } from './test-helpers.js'
+import { createHandler } from '../lib/handler.js'
+import { DATABASE_URL } from '../lib/config.js'
+
+const debug = createDebug('test')
+
+describe('Platform Routes HTTP request handler', () => {
+  /** @type {pg.Pool} */
+  let pgPool
+  /** @type {http.Server} */
+  let server
+  /** @type {string} */
+  let baseUrl
+
+  before(async () => {
+    pgPool = new pg.Pool({ connectionString: DATABASE_URL })
+
+    const handler = createHandler({
+      pgPool,
+      logger: {
+        info: debug,
+        error: console.error,
+        request: debug
+      }
+    })
+
+    server = http.createServer(handler)
+    server.listen()
+    await once(server, 'listening')
+    baseUrl = `http://127.0.0.1:${server.address().port}`
+  })
+
+  after(async () => {
+    server.closeAllConnections()
+    server.close()
+    await pgPool.end()
+  })
+
+  beforeEach(async () => {
+    await pgPool.query('DELETE FROM daily_node_metrics')
+  })
+
+  describe('GET /stations/raw', () => {
+    it('returns daily station metrics for the given date range', async () => {
+      await givenDailyStationMetrics(pgPool, '2024-01-10', 'station1')
+      await givenDailyStationMetrics(pgPool, '2024-01-11', 'station2')
+      await givenDailyStationMetrics(pgPool, '2024-01-12', 'station3')
```

Comment:
Let's enhance the test data to check the following cases:
- Only one station recorded in a day
- More than one station was recorded in a day

```suggestion
      await givenDailyStationMetrics(pgPool, '2024-01-11', 'station2')
      await givenDailyStationMetrics(pgPool, '2024-01-12', 'station2')
      await givenDailyStationMetrics(pgPool, '2024-01-12', 'station3')
```